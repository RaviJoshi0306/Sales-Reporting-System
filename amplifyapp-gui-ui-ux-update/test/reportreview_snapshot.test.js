/**
 * @jest-environment jsdom
 */

import '@testing-library/jest-dom';

describe('Sales Report Review Page', () => {
  test('renders the Sales Report Review header and structure', () => {
    document.body.innerHTML = `
    <header class="navbar navbar-expand-lg navbar-light bg-light">
        <div class="container-fluid d-flex justify-content-between align-items-center">
            <div class="d-flex align-items-center">
                <a href="report_review.html" class="d-flex align-items-center text-decoration-none">
                    <span id="company-logo" class="rounded-pill px-3 py-1 bg-light-blue text-secondary">
                        <img src="https://i.postimg.cc/SKq3ZtJZ/KK-logo.png" alt="Sanyo Steel Co Ltd Logo" style="max-height: 40px; max-width: auto;">
                    </span>
                    <span id="report-review" class="h4 font-weight-bold ml-2" style="color: #00a4b3;">営業報告システム</span>
                </a>
            </div>
            <div class="d-flex align-items-center">
                <div id="language-switch" class="mr-4">
                    <a id="lang-jp" class="text-muted">日本語</a> | 
                    <a id="lang-en" class="text-muted">English</a>
                </div>
                <button id="new-report-btn" class="btn btn-primary mr-3">
                    <i class="fas fa-edit"></i> 新規作成
                </button>
                <div class="dropdown">
                    <button class="btn btn-outline-secondary profile-button" type="button" id="profileButton">
                        <i class="fas fa-user-circle"></i>
                        <span id="salesRepName">Loading...</span>
                        <i class="fas fa-chevron-down dropdown-arrow"></i>
                    </button>
                    <ul class="dropdown-menu" id="dropdownMenu" style="display:none;">
                        <li>
                            <button id="logout" class="dropdown-item text-danger">
                                <i class="fas fa-power-off"></i> ログアウト
                            </button>
                        </li>
                    </ul>
                </div>
            </div>
        </div>
    </header>
    `;

    expect(document.body).toMatchSnapshot();
  });
});
