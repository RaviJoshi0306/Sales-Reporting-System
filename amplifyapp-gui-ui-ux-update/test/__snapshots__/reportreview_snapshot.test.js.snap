// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sales Report Review Page renders the Sales Report Review header and structure 1`] = `
<body>
  
    
  <header
    class="navbar navbar-expand-lg navbar-light bg-light"
  >
    
        
    <div
      class="container-fluid d-flex justify-content-between align-items-center"
    >
      
            
      <div
        class="d-flex align-items-center"
      >
        
                
        <a
          class="d-flex align-items-center text-decoration-none"
          href="report_review.html"
        >
          
                    
          <span
            class="rounded-pill px-3 py-1 bg-light-blue text-secondary"
            id="company-logo"
          >
            
                        
            <img
              alt="Sanyo Steel Co Ltd Logo"
              src="https://i.postimg.cc/SKq3ZtJZ/KK-logo.png"
              style="max-height: 40px; max-width: auto;"
            />
            
                    
          </span>
          
                    
          <span
            class="h4 font-weight-bold ml-2"
            id="report-review"
            style="color: #00a4b3;"
          >
            営業報告システム
          </span>
          
                
        </a>
        
            
      </div>
      
            
      <div
        class="d-flex align-items-center"
      >
        
                
        <div
          class="mr-4"
          id="language-switch"
        >
          
                    
          <a
            class="text-muted"
            id="lang-jp"
          >
            日本語
          </a>
           | 
                    
          <a
            class="text-muted"
            id="lang-en"
          >
            English
          </a>
          
                
        </div>
        
                
        <button
          class="btn btn-primary mr-3"
          id="new-report-btn"
        >
          
                    
          <i
            class="fas fa-edit"
          />
           新規作成
                
        </button>
        
                
        <div
          class="dropdown"
        >
          
                    
          <button
            class="btn btn-outline-secondary profile-button"
            id="profileButton"
            type="button"
          >
            
                        
            <i
              class="fas fa-user-circle"
            />
            
                        
            <span
              id="salesRepName"
            >
              Loading...
            </span>
            
                        
            <i
              class="fas fa-chevron-down dropdown-arrow"
            />
            
                    
          </button>
          
                    
          <ul
            class="dropdown-menu"
            id="dropdownMenu"
            style="display:none;"
          >
            
                        
            <li>
              
                            
              <button
                class="dropdown-item text-danger"
                id="logout"
              >
                
                                
                <i
                  class="fas fa-power-off"
                />
                 ログアウト
                            
              </button>
              
                        
            </li>
            
                    
          </ul>
          
                
        </div>
        
            
      </div>
      
        
    </div>
    
    
  </header>
  
    
</body>
`;
